### PR TITLE
10129: update pauli_list.py evolve method docstring

### LINKED ISSUE #10129: docs typo in `PauliList.evolve`
### Environment

- **Qiskit Terra version**: Online documentation
- **Python version**: 
- **Operating system**: 

### What is happening?

https://qiskit.org/documentation/stubs/qiskit.quantum_info.PauliList.evolve.html

I think the value returned with frame = 'h' should be different from the value returned with frame='s'. Seems like one or two daggers need to move:
image

If we trust the docs for Pauli.evolve, then the error on PauliList.evolve docs is for the default behavior (frame = 'h'), which should instead be `C^dag P C` :
image

### How can we reproduce the issue?

https://qiskit.org/documentation/stubs/qiskit.quantum_info.PauliList.evolve.html

### What should happen?

See above

### Any suggestions?

_No response_

### PR DESCRIPTION
**Pull request by thspreetham98**
### Summary
To fix #10129

### CODE REVIEW

**Review by ShellyGarion — DISMISSED** [review]
LGTM, thank you for fixing this!
@ikkoham - could you please approve and merge this PR?

**Review by 1ucian0 — APPROVED** [review]

File: qiskit/quantum_info/operators/symplectic/pauli_list.py [hunk 1]
```diff
@@ -903,7 +903,7 @@ def _commutes_with_all(self, other, anti=False):
     def evolve(self, other, qargs=None, frame="h"):
         r"""Evolve the Pauli by a Clifford.
 
-        This returns the Pauli :math:`P^\prime = C.P.C^\dagger`.
+        This returns the Pauli :math:`P^\prime = C^\dagger.P.C`.
 
         By choosing the parameter frame='s', this function returns the Schrödinger evolution of the Pauli
```
Comment (1ucian0) [inline qiskit/quantum_info/operators/symplectic/pauli_list.py:908]:
```suggestion
        By choosing the parameter ``frame='s'``, this function returns the Schrödinger evolution of the Pauli
```

**Review by 1ucian0 — APPROVED** [review]

File: qiskit/quantum_info/operators/symplectic/pauli_list.py [hunk 2]
```diff
@@ -914,7 +914,7 @@ def evolve(self, other, qargs=None, frame="h"):
             frame (string): 'h' for Heisenberg or 's' for Schrödinger framework.
```
Comment (1ucian0) [inline qiskit/quantum_info/operators/symplectic/pauli_list.py:914]:
```suggestion
            frame (string): 'h' for Heisenberg (Default ) or 's' for Schrödinger framework.
```

File: qiskit/quantum_info/operators/symplectic/pauli_list.py [hunk 3]
```diff
@@ -914,7 +914,7 @@ def evolve(self, other, qargs=None, frame="h"):
...
 
         Returns:
-            Pauli: the Pauli :math:`C.P.C^\dagger`.
+            Pauli: the Pauli :math:`C^\dagger.P.C`.
```
Comment (1ucian0) [inline qiskit/quantum_info/operators/symplectic/pauli_list.py:917]: Following up on the suggestion in https://github.com/Qiskit/qiskit-terra/issues/10129#issuecomment-1554321331

```suggestion
            Pauli: the Pauli :math:`C^\dagger.P.C` (Schrödinger framework) or the Pauli :math:`C.P.C^\dagger` (Heisenberg framework). 
```

**Review by 1ucian0 — COMMENTED** [review]
Great first PR! Welcome to the dev community.

Some formatting suggestions. If you agree with them, consider adding them in `qiskit.quantum_info.Pauli.evolve`.

**Review by ikkoham — COMMENTED** [review]
The changes are looks good to me and I agree with @1ucian0's suggestions.

**Comment by thspreetham98** [comment]
I have accepted your suggestions. Please let me know if it looks OK now. Thank again!

**Comment by thspreetham98** [comment]
@1ucian0 @ikkoham can you please review?

**Comment by 1ucian0** [comment]
Great! What do you think about doing the same in the Pauli object?

File: qiskit/quantum_info/operators/symplectic/pauli_list.py [hunk 4]
```diff
@@ -903,18 +903,20 @@ def _commutes_with_all(self, other, anti=False):
...
         Args:
             other (Pauli or Clifford or QuantumCircuit): The Clifford operator to evolve by.
             qargs (list): a list of qubits to apply the Clifford to.
-            frame (string): 'h' for Heisenberg or 's' for Schrödinger framework.
+            frame (string): ``'h'`` for Heisenberg (default) or ``'s'`` for Schrödinger framework.
 
         Returns:
-            Pauli: the Pauli :math:`C.P.C^\dagger`.
+            Pauli: the Pauli :math:`C^\dagger.P.C` (Schrödinger framework)
+            or the Pauli :math:`C.P.C^\dagger` (Heisenberg framework).
```
Comment (jakelishman) [inline qiskit/quantum_info/operators/symplectic/pauli_list.py:919]: I never remember which way round the Heisenberg and Schrödinger pictures are, but this text currently seems to disagree with itself.  Lines 909--910 say the Schrödinger frame is $P' = CPC^\dagger$, but the return value says that the Schrödinger frame returns $C^\dagger PC$.

I 100% defer to Shelley and Ikko on which is which, but right now it reads to me that we're inconsistent.

Comment (ShellyGarion) [reply qiskit/quantum_info/operators/symplectic/pauli_list.py:919]: I think that it was mixed in this comment (and should be the other way round):
https://github.com/Qiskit/qiskit-terra/pull/10219#discussion_r1221264651

Comment (thspreetham98) [reply qiskit/quantum_info/operators/symplectic/pauli_list.py]: Hi. I rebased my code with main and now my MR is showing more file changes than expected. Can you help me fix this or shall I just create a new MR?

**Comment by thspreetham98** [comment]
I will be creating a new PR for the same

**Pull request closed without merging**